### ISSUE TITLE
building/dependencies?

### ISSUE THREAD

**Issue by dchmelik** [ev3dev/grx#18]
I maintain grx.SlackBuild at [SlackBuilds.org](http://slackbuilds.org/) (main extra build script repository for Slackware GNU/Linux) so people can install grx.  However, after v2.4.9, there are no building/dependencies instructions.  [SlackBuilds templates](http://slackbuilds.org/templates/cmake-template.SlackBuild) says do the following.
```
mkdir -p build
cd build
  cmake \
    -DCMAKE_C_FLAGS:STRING="$SLKCFLAGS" \
    -DCMAKE_CXX_FLAGS:STRING="$SLKCFLAGS" \
    -DCMAKE_INSTALL_PREFIX=/usr \
    -DLIB_SUFFIX=${LIBDIRSUFFIX} \
    -DMAN_INSTALL_DIR=/usr/man \
    -DCMAKE_BUILD_TYPE=Release ..
  make
  make install/strip DESTDIR=$PKG
cd ..
```
I tried to update grx.SlackBuild to do that, which now says the following.
```
CMake Error at src/CMakeLists.txt:215 (find_package):
  By not providing "FindGLibGenMarshal.cmake" in CMAKE_MODULE_PATH this
  project has asked CMake to find a package configuration file provided by
  "GLibGenMarshal", but CMake did not find one.

  Could not find a package configuration file provided by "GLibGenMarshal"
  with any of the following names:

    GLibGenMarshalConfig.cmake
    glibgenmarshal-config.cmake

  Add the installation prefix of "GLibGenMarshal" to CMAKE_PREFIX_PATH or set
  "GLibGenMarshal_DIR" to a directory containing one of the above files.  If
  "GLibGenMarshal" provides a separate development package or SDK, be sure it
  has been installed.
```
Please document building/dependencies.

**Comment by dlech** [comment]
This fork of GRX is quite different from the 2.x version. All APIs have been renamed and it has been made to use GLib.

You may be more interested in https://www.fgrim.com/mgrx/mgrxpm.htm which is a different fork that is more actively maintained and has stayed much closer to the original GRX 2.x APIs.

If you are still interested in fixing the build error, it looks like the git submodule didn't get checked out.

**Comment by dchmelik** [comment]
Yes, I am interested in fixing, but I (like most SlackBuilds) use stable/normally-numbered before commit/hash-numbered versions (so I downloaded tarball, not using git).  Slackware 15 stable (up-to-date 64-bit n86 (a.k.a. x86)) has the following.
- glib-1.2.10-x86_64-9
- glib2-2.70.3-x86_64-1
